core/indigo-core/molecule/src/molecule_json_saver.cpp

```diff
@@ -1139,12 +1138,81 @@ std::string MoleculeJsonSaver::monomerAlias(const TGroup& tg)
     return alias;
 }
 
+void MoleculeJsonSaver::saveSuperAtomAsTemplate(BaseMolecule& mol, int sa_index, int template_number, JsonWriter& writer)
+{
+    Superatom& sa = (Superatom&)mol.sgroups.getSGroup(sa_index);
```

Comment:
this logic should not be part of JSON saver. Super atoms should be converted to templates while reading from molfile